client/src/pages/Report/CreateReport.tsx

```diff
@@ -0,0 +1,246 @@
+import './CreateReport.scss';
+import HeaderWithIcon from 'components/PageTemplate/components/HeaderWithIcon/HeaderWithIcon.tsx';
+import PageTemplate from 'components/PageTemplate/PageTemplate.tsx';
+import AccuracySection from './AccuracySection/AccuracySection.tsx';
+import DateRangeSection from './DateRangeSection/DateRangeSection.tsx';
+import NotificationSection from './NotificationSection/NotificationSection.tsx';
+import ApplicationSection from './ApplicationSection/ApplicationSection.tsx';
+import NodesSection from './NodesSection/NodesSection.tsx';
+import ActionButton, {ActionButtonColor} from 'components/ActionButton/ActionButton.tsx';
+import {useNavigate, useParams} from 'react-router-dom';
+import {useState, useEffect} from 'react';
+import {NotificationChannel} from './NotificationSection/NotificationSection';
+import {ApplicationDataRow} from './ApplicationSection/ApplicationSection';
+import {NodeDataRow} from './NodesSection/NodesSection';
+import {
+    AccuracyLevel, ClusterUpdateData, ManagmentServiceApiInstance,
+    ReportPost, ReportType, NotificationChannelKind
+} from 'api/managment-service.ts';
+import GeneratedInfoPopup from './GeneratedInfoPopup/GeneratedInfoPopup.tsx';
+import ReportGenerationType from './StateSection/ReportGenerationType.tsx';
+import {dateFromTimestampMs} from 'lib/date.ts';
+import SchedulePeriod, {periodToMilliseconds, schedulePeriodOptions}
+    from './SchedulePeriod/SchedulePeriod.tsx';
+
+const CreateReport = () => {
+    const {id} = useParams<{ id: string }>();
+    const [notificationChannels, setNotificationChannels] = useState<NotificationChannel[]>([]);
+    const [applications, setApplications] = useState<ApplicationDataRow[]>([]);
+    const [nodes, setNodes] = useState<NodeDataRow[]>([]);
+    const [accuracy, setAccuracy] = useState<AccuracyLevel>('HIGH');
+    const [generationType, setGenerationType] = useState<ReportType>('ON DEMAND');
+    const [generationPeriod, setGenerationPeriod] = useState<string>(schedulePeriodOptions.periods[2]);
+    const navigate = useNavigate();
+    const [startDateMs, setStartDateMs] = useState<number>(Date.now());
+    const [endDateMs, setEndDateMs] = useState<number>(Date.now());
+    const [showInfoPopup, setShowInfoPopup] = useState(false);
+
+    const handleDateRangeChange = (startMs: number, endMs: number) => {
+        setStartDateMs(startMs);
+        setEndDateMs(endMs);
+    };
+
+    useEffect(() => {
+        const fetchClusterDetails = async () => {
+            try {
+                const [clusterDetails, runningApplications, runningNodes] = await Promise.all([
```

Comment:
Do we need to wait for all of them to start rendering?
Isn't it possible to fetch them separately?

Reply:
And load each section once given fetch was completed?

Reply:
TODO